### PR TITLE
[R4R] Add surplus and debt auction lot params

### PR DESCRIPTION
**Pull request by karzak**
Adds `SurplusAuctionLot` and `DebtAuctionLot` params to the CDP module. These params allow us to handle the case where, for example, we want to build up a large surplus, and then hold a surplus auction for some small fraction of the surplus. In the previous logic, there was no way to accomplish this, surplus and debt auctions were all or nothing.

### CODE REVIEW

File: x/cdp/keeper/auctions.go [hunk 1]
```diff
@@ -119,7 +123,11 @@ func (k Keeper) RunSurplusAndDebtAuctions(ctx sdk.Context) error {
 	if !surplus.GTE(params.SurplusAuctionThreshold) {
 		return nil
 	}
-	surplusLot := sdk.NewCoin(params.DebtParam.Denom, surplus)
+	surplusLot := sdk.NewCoin(params.DebtParam.Denom, params.SurplusAuctionLot)
+	// sanity check - don't auction more coins than we have available
+	if params.SurplusAuctionLot.GT(surplus) {
+		surplusLot = sdk.NewCoin(params.DebtParam.Denom, surplus)
+	}
```
Comment (fedekunze) [inline x/cdp/keeper/auctions.go:130]:
```suggestion
	surplusLot := sdk.NewCoin(params.DebtParam.Denom, sdk.MinInt
	(surplus, params.SurplusAuctionLot))
```

File: x/cdp/keeper/auctions.go [hunk 2]
```diff
@@ -109,7 +109,11 @@ func (k Keeper) RunSurplusAndDebtAuctions(ctx sdk.Context) error {
 	remainingDebt := k.GetTotalDebt(ctx, types.LiquidatorMacc)
 	params := k.GetParams(ctx)
 	if remainingDebt.GTE(params.DebtAuctionThreshold) {
-		_, err := k.auctionKeeper.StartDebtAuction(ctx, types.LiquidatorMacc, sdk.NewCoin("usdx", remainingDebt), sdk.NewCoin(k.GetGovDenom(ctx), remainingDebt.Mul(sdk.NewInt(dump))), sdk.NewCoin(k.GetDebtDenom(ctx), remainingDebt))
+		debtLot := sdk.NewCoin(k.GetDebtDenom(ctx), params.DebtAuctionLot)
+		debtCoveredByLot := (sdk.NewDecFromInt(params.DebtAuctionLot).Quo(sdk.NewDecFromInt(remainingDebt))).Mul(sdk.NewDecFromInt(remainingDebt)).RoundInt()
```
Comment (fedekunze) [inline x/cdp/keeper/auctions.go:113]: nit, can you add a comment here with the formula? Is it the right one?
`debtAuctionLot*remainingDebt/remainingDebt`

Comment (alexanderbez) [reply x/cdp/keeper/auctions.go:113]: Yah I totally second this 👍

File: x/cdp/spec/06_params.md [hunk 3]
```diff
@@ -8,12 +8,16 @@ The cdp module contains the following parameters:
 | DebtParams                   | DebtParam               | {see below}                        | array of params for each enabled pegged asset                    |
 | GlobalDebtLimit              | coin                    | {"denom":"usdx","amount":"1000"}   | maximum pegged assets that can be minted across the whole system |
 | SavingsDistributionFrequency | string (int)            | "84600"                            | number of seconds between distribution of the savings rate       |
-| CircuitBreaker               | bool                    | false                              | flag to disable user interactions with the system                |
+| GlobalDebtLimit              | coin                    | {"denom":"usdx","amount":"1000"}   | maximum pegged assets that can be minted across the whole system |
```
Comment (alexanderbez) [inline x/cdp/spec/06_params.md:11]: Not sure how Kava renders its website documentation, but you may have to escape (via backticks) any `{}`.

Comment (karzak) [reply x/cdp/spec/06_params.md:11]: Ty for the heads-up, this is indeed causing issues. Will make sure it gets addressed in separate PR.

File: x/cdp/keeper/auctions.go [hunk 4]
```diff
@@ -109,7 +109,12 @@ func (k Keeper) RunSurplusAndDebtAuctions(ctx sdk.Context) error {
 	remainingDebt := k.GetTotalDebt(ctx, types.LiquidatorMacc)
 	params := k.GetParams(ctx)
 	if remainingDebt.GTE(params.DebtAuctionThreshold) {
-		_, err := k.auctionKeeper.StartDebtAuction(ctx, types.LiquidatorMacc, sdk.NewCoin("usdx", remainingDebt), sdk.NewCoin(k.GetGovDenom(ctx), remainingDebt.Mul(sdk.NewInt(dump))), sdk.NewCoin(k.GetDebtDenom(ctx), remainingDebt))
+		debtLot := sdk.NewCoin(k.GetDebtDenom(ctx), params.DebtAuctionLot)
+		// debtAuctionLot * remainingDebt / remainingDebt
```
Comment (fedekunze) [inline x/cdp/keeper/auctions.go:113]: q: why are we multiplying and dividing by `remainingDebt` instead of just using `debtAuctionLot`?

Comment (karzak) [reply x/cdp/keeper/auctions.go:113]: 😂 this formula is totally unnecessary, will remove

**Pull request merged**